### PR TITLE
feat(gha): multi os base64.

### PR DESCRIPTION
**Pull request by Zenithar**
#  Context

Add platform dependent `base64` encoding function.

# Reference(s)

- https://github.com/philips-labs/slsa-provenance-action/issues/146

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -40,8 +40,28 @@ runs:
...
+        base64func() {
+          case ${{ runner.os }} in
+            Linux)
+              echo $1 | base64 -w 0
+              ;;
+            macOS)
+              echo $1 | base64 -b 0
+              ;;
+            Windows)
+              powershell -command "[System.Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes($1))"
```
Comment (marcofranssen) [inline action.yaml:54]: This is not working. Seems the $1 is not seen as a string in this PowerShell function.

**Review by marcofranssen — CHANGES_REQUESTED** [review]
Thanks a lot for this PR. 2 small things to cover.

**Comment by Zenithar** [comment]
I'm currently working on a different approach because by serializing the GithubContext and display the base64 passed to the command line. We can dump the generated token.

image

**Comment by Zenithar** [comment]
I have identified various multiplatform support problems (use tar with zip, etc.)

I will do another PR later. Closing this one.

**Pull request closed without merging**